alacritty_terminal/src/term/mod.rs

```diff
@@ -1440,22 +1442,44 @@ impl<T> Term<T> {
         let text_color = color.text();
 
         // Expand across wide cell when inside wide char or spacer.
-        let buffer_point = self.visible_to_buffer(point);
+        let buffer_point = self.visible_to_buffer(cursor_point);
         let cell = &self.grid[buffer_point.line][buffer_point.col];
         let is_wide = if cell.flags.contains(Flags::WIDE_CHAR_SPACER) {
-            point.col -= 1;
+            cursor_point.col -= 1;
             true
         } else {
             cell.flags.contains(Flags::WIDE_CHAR)
         };
 
-        RenderableCursor {
-            text_color,
+        let buffer_point = self.grid.visible_to_buffer(self.grid.cursor.point);
+
+        let mut inner = RenderableContentInner {
+            term: self,
+            config,
+            selection: self.visible_selection(),
+            cursor_point,
+            cursor_shape,
+            search: RenderableSearch::new(self),
+        };
+
+        let cell = Indexed {
+            inner: &self.grid[buffer_point.line][buffer_point.col],
+            column: cursor_point.col,
+            line: cursor_point.line,
+        };
+        let mut cell = RenderableCell::new(&mut inner, cell);
```

Comment:
Do we create an entire `RenderableContentInner` including selection, config, term and search, then create a cell from it, just to check which color our cursor is supposed to be?

It feels like there's got to be an easier way to figure out a cursor's color than that.

Reply:
Well, all coloring stuff is handles by renderable cell. And we have a lot of fancy logic to get colors, etc, etc. Maybe we can craft something simple, but I'm not sure.

Reply:
Yeah, but you don't need stuff like `bg_alpha`, `column/line`, `zerowidth`, `flags` or `match` detection. Sure the color logic is here, but it just seems like there's also a lot of other logic that we don't really care about here?

I think the biggest problem is just that we kinda construct a full renderable cell here, when we know that the renderable cell will already be created in a different place. It would be nice if we could just access the information from that cell instead of having to build it twice.

Reply:
But the information will be there only when iteration is done, we could pass a mutable reference to a renderable cursor and finish initialization there.

Reply:
At that point we could just return a struct from `renderable_cells` which contains cursor and cells.